frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/utils.ts

```diff
@@ -223,14 +229,74 @@ export const getEdgesFromNodes = (nodes: PipelineMixedNodeModel[]): PipelineEdge
     }),
   ).filter((edgeList) => !!edgeList);
 
+export const getFinallyTaskHeight = (allTasksLength: number, disableBuilder: boolean): number => {
+  return (
+    allTasksLength * NODE_HEIGHT +
+    (allTasksLength - 1) * FINALLY_NODE_VERTICAL_SPACING +
+    (!disableBuilder ? NODE_HEIGHT : 0) +
+    FINALLY_NODE_PADDING * 2
+  );
+};
+
+export const getLastRegularTasks = (regularTasks: PipelineMixedNodeModel[]): string[] => {
+  const runAfters = regularTasks.reduce((acc, { data: { task: { runAfter } } }) => {
+    if (runAfter) {
+      acc.push(...runAfter);
+    }
+    return _.uniq(acc);
+  }, []);
```

Comment:
Not that it really matters, but this is inefficient. `uniq` after the fact.
```suggestion
  const runAfters = _.uniq(regularTasks.reduce((acc, { data: { task: { runAfter } } }) => {
    // more simple? 
    return runAfter ? acc.concat(runAfter) : acc;
  }, []));
```